### PR TITLE
Changes tests to not follow external links

### PR DESCRIPTION
**Pull request by spectranaut**
Fixes issue #1594

Following external links in tests to test key board interactions with links cause many time out related failures in github checks. I replace the four tests to replace the `href` on the dom element with a url fragment to test the keyboard behavior.

### CODE REVIEW

File: test/tests/link_link.js [hunk 1]
```diff
@@ -103,18 +103,24 @@ ariaTest(
       let linkLocator = By.css(ex.linkSelector);
       let linkElement = await t.context.session.findElement(linkLocator);
 
+      // Update url to remove external reference for dependable testing
+      const newUrl = t.context.url + '#test-url-change';
+      await t.context.session.executeScript(
+        function () {
+          let [selector, newUrl] = arguments;
+          document.querySelector(selector).onkeydown = function () {
+            goToLink(event, newUrl);
```
Comment (rwaldron) [inline test/tests/link_link.js:112]: There's no `event` here

Comment (rwaldron) [reply test/tests/link_link.js:112]: I think you can make this whole thing re-usable: 

```js
function onKeydownGoToLink(t, linkSelector, newUrl) {
  await t.context.session.executeScript(
    function () {
      let [selector, newUrl] = arguments;
      document.querySelector(selector).onkeydown = function (event) {
        goToLink(event, newUrl);
      };
    },
    linkSelector,
    newUrl
  )  
}
```

And called like: 

```js
const newUrl = t.context.url + '#test-url-change';
onKeydownGoToLink(t, ex.linkSelector, newUrl);
```

I've looked through the other cases and I think this meets them all.

File: test/tests/menu-button_links.js [hunk 2]
```diff
@@ -270,13 +258,24 @@ ariaTest('"enter" on role="menuitem"', exampleFile, 'menu-enter', async (t) => {
...
+    const newUrl = t.context.url + '#test-url-change';
+    await t.context.session.executeScript(
+      function () {
+        let [selector, index, newUrl] = arguments;
+        document.querySelectorAll(selector)[index].href = newUrl;
+      },
+      ex.menuitemSelector,
+      index,
+      newUrl
+    );
```
Comment (rwaldron) [inline test/tests/menu-button_links.js:271]: My suggestion above could be used to replace this by providing a different selector: 

```js
onKeydownGoToLink(t, `${ex.menuitemSelector}:nth-of-type(${index}n)`, newUrl)
```

For a quick example: 
```js
document.querySelector('body:nth-of-type(1n)') === document.querySelectorAll('body')[0]
```

File: test/tests/treeview_treeview-2a.js [hunk 3]
```diff
@@ -272,21 +272,24 @@ ariaTest(
...
+    // Update url to remove external reference for dependable testing
+    const newUrl = t.context.url + '#test-url-change';
+    await t.context.session.executeScript(
+      function () {
+        let [selector, newUrl] = arguments;
+        document.querySelectorAll(selector)[0].href = newUrl;
+      },
+      ex.linkSelector,
+      newUrl
+    );
```
Comment (rwaldron) [inline test/tests/treeview_treeview-2a.js:284]: Same replacement suggestion as my previous comment

**Review by rwaldron — CHANGES_REQUESTED** [review]

File: test/tests/link_link.js (hunk 1 above)
Comment (spectranaut) [reply test/tests/link_link.js:112]: I appreciate the cleaner code but we can't actually apply this one to the others -- the [link example](https://w3c.github.io/aria-practices/examples/link/link.html) shows how to use aria to communicate something is a link that has onclick and onkeydown handlers, so there is a "goToLink" global function defined in the example page. The others are just regular links so we just need to replace the "href".

**Comment by spectranaut** [comment]
Thanks for the feedback @rwaldron ! I added a helper function for all cases but the link case. It def cleans up the code :)

**Comment by mcking65** [comment]
@spectranaut We should have some documentation about this technique and helper function on our wiki page about writing regression tests.

**Pull request merged**